src/DotNet.Status.Web/TimelineIssueTriageService.cs

```diff
@@ -0,0 +1,286 @@
+                _logger.LogInformation($"buildId: {triageItem.BuildId}, recordId: {triageItem.RecordId}, index: {triageItem.Index}, category: {triageItem.UpdatedCategory}, url: {triageItem.Url}");
+            }
+
+            await IngestTriageItemsIntoKusto(triageItems);
+
+            await gitHubClient.Issue.Comment.Create(issuePayload.Repository.Id, issuePayload.Issue.Number, $"Bot has updated the 'TimelineIssuesTriage' database.\n**PowerBI reports may take up to 24 hours to refresh**\n\nSee {_docLink} for more information.");
+
+            return;
+        }
+
+        private async Task IngestTriageItemsIntoKusto(ICollection<TriageItem> triageItems)
+        {
+            _logger.LogInformation("Entering IngestTriageItemIntoKusto");
+            await KustoHelpers.WriteDataToKustoInMemoryAsync(
+                _kustoIngestClientFactory.GetClient(),
+                _kustoOptions.Value.Database,
+                "TimelineIssuesTriage",
+                _logger,
+                triageItems,
+                b => new[]
+                {
+                    new KustoValue("ModifiedDateTime", b.ModifiedDateTime, KustoDataType.DateTime),
+                    new KustoValue("BuildId", b.BuildId, KustoDataType.Int),
+                    new KustoValue("RecordId", b.RecordId, KustoDataType.Guid),
+                    new KustoValue("Index", b.Index, KustoDataType.Int),
+                    new KustoValue("UpdatedCategory", b.UpdatedCategory, KustoDataType.String),
+                    new KustoValue("Url", b.Url, KustoDataType.String)
+                });
+        }
+
+        private bool IsDuplicate(IList<TriageItem> triageItems, IList<TriageItem> existingIssueItems)
+        {
+            return _logic.IsDuplicate(triageItems, existingIssueItems);
+        }
+
+        private string GetTriageIssueProperty(string propertyName, string body)
+        {
+            return _logic.GetTriageIssueProperty(propertyName, body);
+        }
+
+        private IList<TriageItem> GetTriageItemsProperties(string body)
+        {
+            return _logic.GetTriageItemsProperties(body);
+        }
+
+        public class TimelineIssueTriageLogic
+        {
+            // Search for triage item identifier information, example - "[BuildId=123456,RecordId=0dc87500-1d33-11ea-8b24-4baedbda8954,Index=0]"
+            private static readonly Regex _triageIssueIdentifierRegex = new Regex(@"\[BuildId=(?<buildid>[^,]+),RecordId=(?<recordid>[^,]+),Index=(?<index>[^\]]+)\]\s+\[Category=(?<category>[^\]]+)\]", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
+            // Search for a triage item property, example = "[Category=foo]"
+            private static readonly Regex _triagePropertyRegex = new Regex(@"\[(?<key>[^=]+)=(?<value>[^\]]+)\]", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
+
+            public IList<TriageItem> GetTriageItemsProperties(string body)
```

Comment:
I think that this name could be a bit misleading because it actually returns `TriageItems` not properties. We have another method below `GetTriageIssueProperty` below which actually returns the property. Maybe it could be renamed to `ParseTriageItems` for example or at least drop `Properties` suffix.